### PR TITLE
Check "Sponsor this project" links for github sponsors

### LINKED ISSUE #15: GitHub Sponsors: Use Entities Listed Under "Sponsor this project"
From issue #2:

> Another option would be to use whatever entity/entities are listed under "Sponsor this project" on github. Sometimes it's an org (https://github.com/babel/babel), sometimes a person (or multiple people: https://github.com/sindresorhus/got). Mixing the two might make any analysis of funding numbers even trickier, though.

Is this functionality already captured by the code below?

https://github.com/georgetown-cset/funder-finder/blob/c0473c422a5b6ab6312163e77d9c6b32483893bf/funderfinder/utils/github_sources.py#L26-L43

If so, then I could draw on this functionality with a new script? Or is the functionality you reference above separate from that function?

### PR DESCRIPTION
**Pull request by jmelot**
Closes #15 

See what you think about this implementation. The idea is to, given a link to a github sponsors page listed under "Sponsor this Project", check whether the link corresponds to the project owner or one of the top contributors. If not, we add in a new record to the GitHub Sponsors funding sources with `funding_type` "sponsor_this_project"

(sorry for the huge delay in getting to this btw, we launched a [new product](https://almanac.eto.tech/) and a major new [feature](https://sciencemap.eto.tech/?all_subjects=computer+security&mode=summary) last week and was distracted for a while! Hope all's well on your end.)

### CODE REVIEW

File: tests/sources/test_github_sponsors.py
Comment (jspeed-meyers) [inline tests/sources/test_github_sponsors.py:42]: This tests could eventually break if the number of sponsors changes, which seems possible since, IIRC, `Flux.jl` s a pretty popular project. At the least, consider adding a comment that explains this possibility so some future maintainer, even if it is us, is less surprised.

File: tests/sources/test_github_sponsors.py
Comment (jspeed-meyers) [inline tests/sources/test_github_sponsors.py:45]: Same thing here, please :)

File: tests/sources/test_github_sponsors.py
Comment (jspeed-meyers) [inline tests/sources/test_github_sponsors.py:49]: This one is safe though, for now :)

File: funderfinder/sources/github_sponsors.py
Comment (jspeed-meyers) [inline funderfinder/sources/github_sponsors.py:59]: Should `:return: Dict of funding stats` now be `:return: number of org funders`?

**Review by jspeed-meyers — APPROVED** [review]
Nice work. Gosh, you rock.

Minor requests related to documentation. Once you've done that, merge away!

File: tests/sources/test_github_sponsors.py
Comment (jmelot) [reply tests/sources/test_github_sponsors.py:42]: Ah good point, will be done in next commit!

File: funderfinder/sources/github_sponsors.py
Comment (jmelot) [reply funderfinder/sources/github_sponsors.py:59]: Good point, will be done in next commit!

**Comment by jmelot** [comment]
Thanks for the comments, they should all be addressed now!

**Pull request merged**
